packages/plugin-github/src/plugins/createPullRequest.ts

```diff
@@ -0,0 +1,170 @@
+}
+
+export const createPullRequestAction: Action = {
+    name: "CREATE_PULL_REQUEST",
+    similes: ["CREATE_PR"],
+    description: "Create a pull request",
+    validate: async (runtime: IAgentRuntime) => {
+        // Check if all required environment variables are set
+        const token = !!runtime.getSetting("GITHUB_API_TOKEN");
+
+        return token;
+    },
+    handler: async (runtime: IAgentRuntime, message: Memory, state: State, options: any, callback: HandlerCallback) => {
+        elizaLogger.log("Composing state for message:", message);
+        if (!state) {
+            state = (await runtime.composeState(message)) as State;
+        } else {
+            state = await runtime.updateRecentMessageState(state);
+        }
+
+        const context = composeContext({
+            state,
+            template: createPullRequestTemplate,
+        });
+
+        const details = await generateObjectV2({
+            runtime,
+            context,
+            modelClass: ModelClass.SMALL,
+            schema: CreatePullRequestSchema,
+        });
+
+        if (!isCreatePullRequestContent(details.object)) {
+            throw new Error("Invalid content");
+        }
+
+        const content = details.object as CreatePullRequestContent;
+        if (!content.owner) {
+            callback(
+                {
+                    text: "Missing github owner. Please provide the owner of the GitHub repository.",
+                },
+                [],
+            );
+            return;
+        }
+        if (!content.repo) {
+            callback(
+                {
+                    text: "Missing github repo. Please provide the name of the GitHub repository.",
+                },
+                [],
+            );
+            return;
+        }
+        if (!content.branch) {
+            callback(
+                {
+                    text: "Missing github branch. Please provide the branch of the GitHub repository.",
+                },
+                [],
+            );
+            return;
+        }
+
+        elizaLogger.info("Creating a pull request...");
+
+        const repoPath = path.join(
+            process.cwd(),
+            ".repos",
+            content.owner,
+            content.repo,
+        );
+
+        await createNewBranch(repoPath, content.branch);
+
+        // TODO: write files to the repository
+
+        // TODO: commit and push changes
+
+        // TODO: create a pull request
+
+        elizaLogger.info("Pull request created successfully!");
+
+        callback(
+            {
+                text: "Pull request created successfully!",
+                attachments: [],
+            }
+        );
+    },
+    examples: [
+        [
+            {
+                user: "{{agentName}}",
+                content: {
+                    text: "Pull request created successfully!",
+                    action: "INITIALIZE_REPOSITORY",
+                },
+            },
+        ],
+    ],
```

Comment:
Should have the exchange so need a {{user}} example of asking to create a PR see my PR for context